msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/IBroker.java

```diff
@@ -67,11 +67,17 @@ default IAuthenticationResult parseBrokerAuthResult(String authority, String idT
             if (idToken != null) {
                 builder.idToken(idToken);
                 if (accountId != null) {
-                    String idTokenJson =
-                            JWTParser.parse(idToken).getParsedParts()[1].decodeToString();
+                    String idTokenJson;
+
+                    try {
+                        idTokenJson = new String(Base64.getDecoder().decode(idToken.split("\\.")[1]), StandardCharsets.UTF_8);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        throw new MsalServiceException("Error parsing ID token, missing payload section. Ensure that the ID token is following the JWT format.",
```

Comment:
It's not a service exception. Service exceptions are eSTS error messages packaged as exceptions. It's a client exception.

Reply:
Good point, this was an error message added in PR https://github.com/AzureAD/microsoft-authentication-library-for-java/pull/894 and just copied here. I changed it to a client exception in the latest commit.